python/python/lance/dataset.py

```diff
@@ -2762,6 +2788,118 @@ def drop(
         _Dataset.drop(str(base_uri), storage_options, ignore_not_found=ignore_not_found)
 
 
+class SqlQuery:
+    """
+    An executable SQL query.
+
+    This is created by calling :meth:`SqlQueryBuilder.build`.
+    """
+
+    def __init__(self, query):
+        self._query = query
+
+    def to_list(self) -> list[pa.RecordBatch]:
```

Comment:
nit: `to_list` returning record batches and `to_batches` returning a reader seems a bit confusing, why not just use the same name as the rust ones?